teamcity: failed tests on master: acceptance/TestDockerCLI, acceptance/TestDockerCLI/test_missing_log_output.tcl

The following tests appear to have failed:

[#354422](https://teamcity.cockroachdb.com/viewLog.html?buildId=354422):

```
--- FAIL: acceptance/TestDockerCLI (203.340s)
test_log_scope.go:78: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerCLI291007915
test_log_scope.go:61: use -show-logs to present logs inline
--- FAIL: acceptance/TestDockerCLI/test_missing_log_output.tcl (34.690s)
null

```
Please assign, take a look and update the issue accordingly.

Hmm, doubt this is something that test wanted. Unfortunately, I think the artifacts have since been deleted (or I'm exceptionally bad at finding them).

``
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd73149]

goroutine 219 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc420744090, 0x2afff40, 0xc421058060)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:183 +0x131
panic(0x1ab0b00, 0x2ab91c0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*Progress).maybeUpdate(...)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/progress.go:116
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).appendEntry(0xc4206fe000, 0xc420eed270, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:542 +0xd9
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).becomeLeader(0xc4206fe000)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:643 +0x287
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).campaign(0xc4206fe000, 0x1cac40d, 0x10)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:666 +0x668
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).Step(0xc4206fe000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:776 +0x33a
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*RawNode).Campaign(0xc42005d9c0, 0xc42079fc50, 0x3)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/rawnode.go:140 +0x71
[17:59:59]	[Test Output]
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroupLocked(0xc4208af000, 0x1d83200, 0x1d82d28, 0xc420fdc270, 0x30)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:548 +0x380
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).maybeInitializeRaftGroup(0xc4208af000, 0x2afff80, 0xc420fdc270)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1750 +0x185
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Send(0xc4208af000, 0x2afff80, 0xc420fdc270, 0x14e586725cdb65be, 0x0, 0x0, 0x0, 0xa, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1778 +0x194
github.com/cockroachdb/cockroach/pkg/storage.(*pendingLeaseRequest).requestLeaseAsync.func1(0x2afff40, 0xc421058060)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_range_lease.go:213 +0x4da
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc420744090, 0x2afff40, 0xc421058060, 0xc42025e440, 0x35, 0x2b0d4e0, 0xc420170ca0, 0xc42068c360)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:269 +0xe6
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:264 +0x133
expect: does "623d4e7:26257\r\nI170918 17:57:56.112687 36 storage/node_liveness.go:325  [n1] starting liveness heartbeat\r\nW170918 17:57:56.112704 36 sql/jobs/registry.go:156  [n1] unable to get node liveness: node not in the liveness table\r\nI170918 17:57:56.112737 36 rpc/context.go:346  [n1] dialing 43455623d4e7:26257\r\nI170918 17:57:56.112849 36 sqlmigrations/migrations.go:316  [n1] trying to get the list of completed migrations\r\nI170918 17:57:56.112930 36 kv/range_iter.go:183  [n1] range descriptor lookup failed: the descriptor for the first range is not available via gossip\r\nI170918 17:57:56.113206 134 kv/range_iter.go:183  [hb,n1] range descriptor lookup failed: the descriptor for the first range is not available via gossip\r\nI170918 17:57:56.115712 150 kv/range_iter.go:183  [ts-poll,n1] range descriptor lookup failed: the descriptor for the first range is not available via gossip\r\nI170918 17:57:56.115784 151 kv/range_iter.go:183  [summaries,n1] range descriptor lookup failed: the descriptor for the first range is not available via gossip\r\nI170918 17:57:56.126873 31 storage/replica.go:4309  [n1,s1,r1/1:/{Min-System/}] applying command with forced error: storage/replica.go:4170: no-op on empty Raft entry\r\nI170918 17:57:56.127389 115 storage/replica.go:5164  [n1,s1,r1/1:/{Min-System/}] gossiping sentinel from store 1, r1\r\nI170918 17:57:56.127462 115 storage/replica.go:5170  [n1,s1,r1/1:/{Min-System/}] gossiping first range from store 1, r1: [(n1,s1):1]\r\nI170918 17:57:56.127875 122 kv/dist_sender.go:239  [n1] gossiped first range descriptor: [(n1,s1):1]\r\nI170918 17:57:56.144978 32 storage/replica.go:4309  [n1,s1,r3/1:/System/NodeLiveness{-Max}] applying command with forced error: storage/replica.go:4170: no-op on empty Raft entry\r\nI170918 17:57:56.158016 160 storage/intent_resolver.go:177  skipping PushTxn for 01090c30-f97d-4307-8bb8-74e17591d874; attempt already in flight\r\nI170918 17:57:56.158569 134 internal/client/db.go:548  [n1,hb] failed batch: unexpected value: raw_bytes:"\252\300\212\230\003\010\001\020\001\032\014\010\374\273\261\201\300\316\341\362\024\020\000" timestamp:<wall_time:1505757473702539008 logical:0 > \r\nI170918 17:57:56.158594 134 internal/client/txn.go:968  [n1,hb] failed batch: unexpected value: raw_bytes:"\252\300\212\230\003\010\001\020\001\032\014\010\374\273\261\201\300\316\341\362\024\020\000" timestamp:<wall_time:1505757473702539008 logical:0 > \r\nI170918 17:57:56.158694 134 storage/node_liveness.go:355  [n1,hb] heartbeat failed on epoch increment; retrying\r\nI170918 17:57:56.166210 33 storage/replica.go:4309  [n1,s1,r7/1:/Table/{SystemCon\u2026-11}] applying command with forced error: storage/replica.go:4170: no-op on empty Raft entry\r\nI170918 17:57:56.166531 33 storage/replica_proposal.go:453  [n1,s1,r7/1:/Table/{SystemCon\u2026-11}] new range lease repl=(n1,s1):1 start=0.000000000,0 epo=1 pro=1505757476.145643220,0 following repl=(n1,s1):1 start=0.000000000,0 exp=1505757482.694435962,0 pro=1505757473.694477561,0\r\nI170918 17:57:56.167463 43 storage/queue.go:593  [split,n1,s1,r10/1:/{Table/13-Max}] processing replica\r\nE170918 17:57:56.167778 219 util/log/crash_reporting.go:114  [split,n1,s1,r10/1:/{Table/13-Max}] a panic has occurred!\r\npanic: runtime error: invalid memory address or nil pointer dereference [recovered]\r\n\tpanic: runtime error: invalid memory address or nil pointer dereference\r\n[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd73149]\r\n\r\ngoroutine 219 [running]:\r\ngithub.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc420744090, 0x2afff40, 0xc421058060)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:183 +0x131\r\npanic(0x1ab0b00, 0x2ab91c0)\r\n\t/usr/local/go/src/runtime/panic.go:491 +0x283\r\ngithub.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*Progress).maybeUpdate(...)\r\n\t/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/progress.go:116\r\ngithub.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).appendEntry(0xc4206fe000, 0xc420eed270, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:542 +0xd9\r\ngithub.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).becomeLeader(0xc4206fe000)\r\n\t/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:643 +0x287\r\ngithub.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).campaign(0xc4206fe000, 0x1cac40d, 0x10)\r\n\t/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:666 +0x668\r\ngithub.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).Step(0xc4206fe000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:776 +0x33a\r\ngithub.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*RawNode).Campaign(0xc42005d9c0, 0xc42079fc50, 0x3)\r\n\t/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/rawnode.go:140 +0x71\r\ngithub.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroupLocked(0xc4208af000, 0x1d83200, 0x1d82d28, 0xc420fdc270, 0x30)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:548 +0x380\r\ngithub.com/cockroachdb/cockroach/pkg/storage.(*Replica).maybeInitializeRaftGroup(0xc4208af000, 0x2afff80, 0xc420fdc270)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1750 +0x185\r\ngithub.com/cockroachdb/cockroach/pkg/storage.(*Replica).Send(0xc4208af000, 0x2afff80, 0xc420fdc270, 0x14e586725cdb65be, 0x0, 0x0, 0x0, 0xa, 0x0, 0x0, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1778 +0x194\r\ngithub.com/cockroachdb/cockroach/pkg/storage.(*pendingLeaseRequest).requestLeaseAsync.func1(0x2afff40, 0xc421058060)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_range_lease.go:213 +0x4da\r\ngithub.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc420744090, 0x2afff40, 0xc421058060, 0xc42025e440, 0x35, 0x2b0" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
expect: set expect_out(spawn_id) "exp5"
```

--- a/pkg/cli/interactive_tests/test_missing_log_output.tcl
+++ b/pkg/cli/interactive_tests/test_missing_log_output.tcl
@@ -87,7 +87,11 @@ start_test "Test that quit does not show INFO by defaults with --logtostderr"
 # that the default logging level is WARNING, so that no INFO messages
 # are printed between the marker and the (first line) error message
 # from quit. Quit will error out because the server is already stopped.
-send "echo marker; $argv quit --logtostderr\r"
+#
+# Since quit is likely to actually emit warnings, but we are just
+# interested in the absence of INFO messages, remove the warnings from
+# the output.
+send "echo marker; $argv quit --logtostderr | grep -v '^W'\r"
 eexpect "marker\r\nError"
 eexpect ":/# "
 end_test

